graalpython/com.oracle.graal.python/src/com/oracle/graal/python/util/PFunctionArgsFinder.java

```diff
@@ -0,0 +1,70 @@
+package com.oracle.graal.python.util;
+
+import com.oracle.graal.python.nodes.PRootNode;
+import com.oracle.graal.python.nodes.argument.ReadArgumentNode;
+import com.oracle.graal.python.nodes.frame.WriteIdentifierNode;
+import com.oracle.graal.python.runtime.interop.InteropArray;
+import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
+import com.oracle.truffle.api.interop.InteropLibrary;
+import com.oracle.truffle.api.interop.TruffleObject;
+import com.oracle.truffle.api.library.ExportLibrary;
+import com.oracle.truffle.api.library.ExportMessage;
+import com.oracle.truffle.api.nodes.Node;
+import com.oracle.truffle.api.nodes.NodeUtil;
+import com.oracle.truffle.api.nodes.RootNode;
+
+import java.util.LinkedList;
+import java.util.List;
+
+public class PFunctionArgsFinder {
+    private final RootNode rootNode;
+
+    public PFunctionArgsFinder(Node node) {
+        this.rootNode = findRootNode(node);
+        assert rootNode instanceof PRootNode;
+    }
+
+    private static RootNode findRootNode(Node node) {
+        Node n = node;
```

Comment:
The `Node` class has a `getRootNode` method, which returns `null`, if no RootNode is available. I would suggest to use that one instead of bringing your own like this.